### PR TITLE
Add Timeline Events

### PR DESCRIPTION
**Pull request by itaivaks**

### CODE REVIEW

File: lib/hubspot/timeline.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module Hubspot
+	class Timeline
```
Comment (adume) [inline lib/hubspot/timeline.rb]: what's with the indentation here?

File: lib/hubspot/connection.rb [hunk 2]
```diff
@@ -72,6 +80,33 @@ def generate_url(path, params={}, options={})
         base_url + path + query
       end
 
+      def generate_url_using_oauth(path, params={}, options={})
```
Comment (adume) [inline lib/hubspot/connection.rb]: Can you try using the `generate_url` method and pass `hapikey: false` in the options arg?

File: lib/hubspot/connection.rb [hunk 3]
```diff
@@ -30,6 +30,14 @@ def put_json(path, opts)
         response.parsed_response
       end
 
+      def put_json_with_token(path, token, opts)
```
Comment (adume) [inline lib/hubspot/connection.rb]: Let's merge this function with `put_json` and allow it (put_json) to take either an extra param or an extra opt to include additional headers.

File: lib/hubspot/connection.rb [hunk 4]
```diff
@@ -23,8 +23,13 @@ def post_json(path, opts)
...
       def put_json(path, opts)
-        url = generate_url(path, opts[:params])
-        response = put(url, body: opts[:body].to_json, headers: { 'Content-Type' => 'application/json' }, format: :json)
+	      if opts[:token].present?
+		      url = generate_url(path, opts[:params], hapikey: false)
+		      response = put(url, body: opts[:body].to_json, headers: { 'Content-Type' => 'application/json', 'Authorization' => "Bearer #{opts[:token]}" }, format: :json)
+	      else
+		      url = generate_url(path, opts[:params])
+		      response = put(url, body: opts[:body].to_json, headers: { 'Content-Type' => 'application/json' }, format: :json)
+	      end
```
Comment (Mihail-K) [inline lib/hubspot/connection.rb]: Why is this indented so much? I think you used tabs instead of spaces here.

**Pull request merged**